fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java

```diff
@@ -228,7 +229,12 @@ public List<Partition> getPartitionsByNames(String dbName, String tblName, List<
             StarRocksConnectorException connectionException = null;
             try {
                 client = getClient();
-                partitions = client.hiveClient.getPartitionsByNames(dbName, tblName, partitionNames);
+                for (int start = 0; start < partitionNames.size(); start += Config.max_hive_partitions_per_rpc) {
```

Comment:
you could split partitions in HiveMetastore.java

Reply:
done